usaspending_api/api_contracts/contracts/v2/autocomplete/awarding_agency_office.md

```diff
@@ -0,0 +1,72 @@
+FORMAT: 1A
+HOST: https://api.usaspending.gov
+
+# Awarding Agency and Office Autocomplete [/api/v2/autocomplete/awarding_agency_office/]
+
+This endpoint can be used to autocomplete Awarding Agency and Office searches on the Advanced Search page. It will return
+any agencies, sub-agencies, and offices that match the search text. Additionally, any matching agencies will include associated
+offices and sub-agencies, any matching sub-agencies will include associated agencies and offices, and any matching offices will include
+associated agencies and sub-agencies.
+
+## POST
+
+This route sends a request to the backend to retrieve awarding agencies and offices matching the specified search text.
+
++ Request (application/json)
+    + Schema
+
+            {
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "type": "object"
+            }
+
+    + Attributes (object)
+        + `limit` (optional, number)
+        + `search_text` (required, string)
+    + Body
+
+            {
+                "search_text": "Defense"
+            }
+
++ Response 200 (application/json)
+    + Attributes (object)
+        + `results` (required, array[AwardingAgencyOfficeMatchObject], fixed-type)
+
+# Data Structures
+
+## AwardingAgencyOfficeMatchObject (object)
++ `toptier_agency` (required, object)
+    + `toptier_code` (required, string)
+    + `abbreviation` (required, string, nullable)
+    + `name` (required, string)
+    + `subtier_agency` (required, array[SubAgencyChildObject])
+    + `office` (required, array[OfficeChildObject])
```

Comment:
If the keyword text matches a toptier agency, why do we not want to display the associated sub-tier agencies?

Reply:
Look at the AC here,

> This means that when an entity such as a Parent Agency is searched for, it’s children will also appear in the results.


I'm a wrong to interpret it as conflicting with your statement "if the input text matches on a toptier_agency, we'll only need to display the toptier_agency itself."

Reply:
Based on the latest guidance from Marcy. I believe this API design could work. 
> All subagencies will include their toptier agency as a subheading to their label, and all offices will include both their subagency and their toptier agency’s names as subheadings.
When someone completes a search in this filter, we want the ability to match that search string to each filter label and any subheadings it may have.

One tricky thing is results may appear at both the top level (directly matched on entity name) and in the list of children. So the frontend will have to be careful how those results are handled, I'm good with the design though since they've already approved.

Reply:
Have you considered making those list of children have plural names though. `subtier_agencies` and `offices`?